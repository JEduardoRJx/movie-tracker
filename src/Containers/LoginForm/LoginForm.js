import React, { Component } from 'react'
import './LoginForm.scss'
import { Link } from 'react-router-dom'


class LoginForm extends Component {
  constructor() {
    super();
    this.state = {

    }
  }

  render() {
    return (
      <section className='form-section'>
        <form>
          <input type='text' placeholder='Enter User Name'/>
          <input type='text'  placeholder='Enter Password'/>
          <Link to='/'>
            <button className='login-btn' type='button'>Login</button>
          </Link>
        </form>
      </section>
    )
  } 
}

export default LoginForm;