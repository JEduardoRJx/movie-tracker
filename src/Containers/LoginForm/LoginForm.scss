.form-section {
  height: 100vh;
  width: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: #282c34
}

form {
  width: 30%;
  height: 200px;
  display: flex;
  flex-direction: column;
  justify-content: space-around;
}

input {
  height: 30px;
  font-size: 1.5em;
  text-align: center;
  border-radius: 5px;
  border: none;
}

.login-btn,
.create-account-btn {
  height: 30px;
  font-size: 1.5em;
  width: 100%;
  border-radius: 5px;
  border: none;
  &:hover:enabled  {
    color: white;
    background-color: #BC8F8F;
    cursor: pointer;
  }
}