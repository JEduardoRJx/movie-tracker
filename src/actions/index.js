export const addMovies = movies => ({
	type: 'ADD_MOVIES',
	movies
});

export const throwError = errorMsg => ({
  type: 'THROW_ERROR',
  errorMsg
})

export const setUser = user => ({
	type: 'SET_USER',
	user
})

export const toggleFavorite = id => ({
	type: 'FAV_MOVIE',
	id
})

export const getFavs = favs => ({
  type: 'GET_FAVS',
  favs
})
