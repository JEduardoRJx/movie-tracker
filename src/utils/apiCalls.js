const baseUrl = 'https://api.themoviedb.org/3/movie/now_playing?';
// const apiKey = '1b20ae1afe685b2871c8d94218f89eba';
const apiKey2 = 'cee1e60becdb4297de68233fbef2f560';

export const fetchMovies = async () => {
  const url = `${baseUrl}api_key=${apiKey2}&language=en-US`
  const response = await fetch(url)
    if (!response.ok) {
      throw Error("There was an issue retrieving your movies. Please try again.");
    }
  let movies = await response.json();
  movies = movies.results.map(movie => {
    const { backdrop_path, genre_ids, id, overview, poster_path, release_date, title, vote_average } = movie
    return { backdrop_path, genre_ids, id, overview, poster_path, release_date, title, vote_average, isFavorite: false }
  })
  return movies
};

export const postUser = async user => {
  const url = 'http://localhost:3001/api/v1/login';
  const options = {
    method: 'POST',
    body: JSON.stringify({
      email: user.email,
      password: user.password
    }),
    headers: {
      'Content-Type': 'application/json'
    }
  }
  const response = await fetch(url, options);
  if (!response.ok && response.status !== 401) {
    throw Error('Sorry, unable to retrieve your account. Try again later.')
  }
  const foundUser = await response.json();
  if (response.status === 401) {
    throw new Error('Username or password incorrect')
  }
  return foundUser;
}

export const addUser = async user => {
  const url = 'http://localhost:3001/api/v1/users';
  const options = {
    method: 'POST',
    body: JSON.stringify({
      name: user.name,
      email: user.email,
      password: user.password
    }),
    headers: {
      'Content-Type': 'application/json'
    }
  }
  const response = await fetch(url, options);
    if (!response.ok && response.status !== 500) {
     throw Error('Sorry, unable to create your account. Try again later.')
  }
  const newUser = await response.json();
    if (response.status === 500) {
      throw new Error('There is already an account with this email. Go to login or use another email address.')
  }
  return newUser;
}

export const addFavorite = async (userId, movie) => {
  const url = `http://localhost:3001/api/v1/users/${userId}/moviefavorites`;
  const options = {
    method: 'POST',
    body: JSON.stringify({
      movie_id: movie.id, 
      title: movie.title, 
      poster_path: movie.poster_path, 
      release_date: movie.release_date, 
      vote_average: movie.vote_average, 
      overview: movie.overview
    }),
    headers: {
      'Content-Type': 'application/json'
    }
  }
  const response = await fetch(url, options)
  if(!response.ok) {
    throw Error('Unable to add movie as favorite. Try again later.')
  }
  const addedFav = await response.json();
  return addedFav;
}

export const getFavorites = async userId => {
  const url = `http://localhost:3001/api/v1/users/${userId}/moviefavorites`;
  const response = await fetch(url)
  if(!response.ok) {
    throw Error('Unable to retrieve your favorite movies as this time. Please try again later')
  }
  const favoriteMovies = await response.json()
  return favoriteMovies
}

export const removeFavorite = async (userId, favId) => {
  const url = `http://localhost:3001/api/v1/users/${userId}/moviefavorites/${favId}`;
  const options = {
    method: 'DELETE',
    headers: {
      'Content-Type': 'application/json'
    }
  }
  const response = await fetch(url, options)
  if(!response.ok) {
    throw Error('Unable to remove this movie as a favorite at this time. Please try again later')
  }
}