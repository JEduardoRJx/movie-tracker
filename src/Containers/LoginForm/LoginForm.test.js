import React from 'react';
import { shallow } from 'enzyme';
import { LoginForm, mapStateToProps, mapDispatchToProps } from './LoginForm';
import * as actions from '../../actions';
import { postUser, getFavorites } from '../../utils/apiCalls';
jest.mock('../../utils/apiCalls');


describe('LoginForm', () => {
  let wrapper;


  beforeEach(() => {
    wrapper = shallow(<LoginForm 

      />)
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })


});