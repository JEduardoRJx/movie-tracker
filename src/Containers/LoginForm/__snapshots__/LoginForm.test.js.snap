// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm should match the snapshot 1`] = `
<section
  className="form-section"
>
  <form>
    <input
      name="email"
      onChange={[Function]}
      placeholder="Enter Email"
      type="text"
      value=""
    />
    <input
      name="password"
      onChange={[Function]}
      placeholder="Enter Password"
      type="text"
      value=""
    />
    <button
      className="login-btn"
      disabled={true}
      onClick={[Function]}
      type="button"
    >
      Login
    </button>
    <Link
      to="/createaccount"
    >
      <button
        className="create-account-btn"
        type="button"
      >
        Create Account
      </button>
    </Link>
  </form>
</section>
`;
